File: zad3.cs
using System;

class Program {
    static int SumOfDigits(int num) { // нахождение суммы чисел
        int sum = 0;
        while (num != 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }

    static void Main() {
        // ввод чисел
        int n;
        do {
            Console.Write("Введите кол-во чисел(Больше 0): ");
        } while (!int.TryParse(Console.ReadLine(), out n) || n <= 0);
        List<double> nums = new List<double>(n);
        Console.Write("Введите числа: ");
        for (int i = 0; i < n; ++i) {
            nums.Add(double.Parse(Console.ReadLine()));
        }

        Console.Write("Числа: ");
        foreach (var num in nums) { // перебор каждого числа
            if (SumOfDigits((int)num) < 10) { // проверка условия
                Console.Write(num + " "); // вывод подходящих чисел
            }
        }
    }
}

File: zad3.rs
use std::io;
use std::io::Write;

fn sum_of_digits(mut num: i32) -> i32 { // нахождение суммы цифр
    let mut sum = 0;
    while num != 0 {
        sum += num % 10;
        num /= 10;
    }
    sum
}

fn main() {
    // ввод чисел
    let mut n = String::new();
    loop {
        print!("Введите кол-во чисел (Больше 0): ");
        io::stdout().flush().unwrap();
        n.clear();
        io::stdin().read_line(&mut n).unwrap();
        let n: i32 = match n.trim().parse() {
            Ok(num) if num > 0 => num,
            _ => continue,
        };
        let mut nums = Vec::with_capacity(n as usize);
        print!("Введите числа: ");
        io::stdout().flush().unwrap();
        for _ in 0..n {
            let mut num = String::new();
            io::stdin().read_line(&mut num).unwrap();
            let num: f64 = num.trim().parse().unwrap();
            nums.push(num);
        }
        
        print!("Числа: ");
        for num in &nums { // перебор чисел
            if sum_of_digits(*num as i32) < 10 { // проверка условия
                print!("{} ", num); // ввод подходящих чисел
            }
        }
        break;
    }
}